sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Listeners/BlobTargetScalerProvider.cs

```diff
@@ -0,0 +1,114 @@
+
+        public BlobTargetScalerProvider(IServiceProvider serviceProvider, TriggerMetadata triggerMetadata)
+        {
+            AzureComponentFactory azureComponentFactory = null;
+            if ((triggerMetadata.Properties != null) && (triggerMetadata.Properties.TryGetValue(nameof(AzureComponentFactory), out object value)))
+            {
+                azureComponentFactory = value as AzureComponentFactory;
+            }
+            else
+            {
+                azureComponentFactory = serviceProvider.GetService<AzureComponentFactory>();
+            }
+            IConfiguration configuration = serviceProvider.GetService<IConfiguration>();
+            AzureEventSourceLogForwarder logForwarder = serviceProvider.GetService<AzureEventSourceLogForwarder>();
+            var factory = serviceProvider.GetService<ILoggerFactory>();
+            BlobMetadata blobMetadata = JsonConvert.DeserializeObject<BlobMetadata>(triggerMetadata.Metadata.ToString());
+            BlobServiceClientProvider blobServiceClientProvider = new BlobServiceClientProvider(configuration, azureComponentFactory, logForwarder, factory.CreateLogger<BlobServiceClient>());
+            BlobServiceClient blobServiceClient = blobServiceClientProvider.Get(blobMetadata.Connection, serviceProvider.GetRequiredService<INameResolver>());
+            _targetScaler = new ZeroToOneTargetScaler(triggerMetadata.FunctionName, blobServiceClient, blobMetadata.Path, factory);
+        }
+
+        public ITargetScaler GetTargetScaler()
+        {
+            return _targetScaler;
+        }
+
+        internal class ZeroToOneTargetScaler : ITargetScaler
+        {
+            private readonly TargetScalerDescriptor _targetScalerDescriptor;
+            private readonly Lazy<Task<BlobLogListener>> _blobLogListener;
+            private readonly ILogger _logger;
+            private readonly string _containerName;
+            private StorageAnalyticsLogEntry _latestPositiveEntry = null;
+
+            public ZeroToOneTargetScaler(string functionId, BlobServiceClient blobServiceClient, string blobPath, ILoggerFactory loggerFactory)
+            {
+                _targetScalerDescriptor = new TargetScalerDescriptor(functionId);
+                _blobLogListener = new(() => BlobLogListener.CreateAsync(
+                    blobServiceClient,
+                    loggerFactory.CreateLogger<BlobListener>(),
+                    CancellationToken.None));
+                _logger = loggerFactory.CreateLogger<ZeroToOneTargetScaler>();
+                int separatorIndex = blobPath.IndexOf("/", StringComparison.OrdinalIgnoreCase);
+                if (separatorIndex > 0)
+                {
+                    _containerName = blobPath.Substring(0, separatorIndex);
+                }
+                else
+                {
+                    throw new ArgumentException($"The blob path '{blobPath}' is not valid. It should be in format '<container-name>/<path-to-blob>'.");
```

Comment:
As per the runtime validation we do, paths can be of multiple forms https://github.com/Azure/azure-sdk-for-net/blob/a3519be5522118aa8ccae84fe9e27f11c011d752/sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/BlobNameValidationAttribute.cs#L31. E.g. can just be a container name. Looks like you'll throw in that case?

Need to be sure this new code is handling all the existing supported blob paths.